commit f96d9f9d2e3ba39352035e6ac26463243484d404
Author: Sebastian Kemper <sebastian_ml@gmx.net>
Date:   Sun Jan 13 19:25:52 2019 +0100

    Setup .pc file to allow use for cross-compiling
    
    The Makefile is currently creating the pkg-config file using static lib
    and include dir statements. Change that so that projects that
    cross-compile hiredis can use pkg-config to setup other programs
    depending on it.
    
    Note: these projects (like OpenWrt) call pkg-config with arguments to
    overwrite some variables in the .pc file, namely:
    
    --define-variable=prefix=<...>
    --define-variable=exec_prefix=<...>
    
    Signed-off-by: Sebastian Kemper <sebastian_ml@gmx.net>

diff --git a/Makefile b/Makefile
index 07b8a83..14d21de 100644
--- a/Makefile
+++ b/Makefile
@@ -166,8 +166,8 @@ $(PKGCONFNAME): hiredis.h
 	@echo "Generating $@ for pkgconfig..."
 	@echo prefix=$(PREFIX) > $@
 	@echo exec_prefix=\$${prefix} >> $@
-	@echo libdir=$(PREFIX)/$(LIBRARY_PATH) >> $@
-	@echo includedir=$(PREFIX)/$(INCLUDE_PATH) >> $@
+	@echo libdir=\$${exec_prefix}/$(LIBRARY_PATH) >> $@
+	@echo includedir=\$${prefix}/$(INCLUDE_PATH) >> $@
 	@echo >> $@
 	@echo Name: hiredis >> $@
 	@echo Description: Minimalistic C client library for Redis. >> $@
